src/main/java/seedu/address/logic/parser/patient/DeletePatientCommandParser.java

```diff
@@ -31,17 +31,17 @@ public DeletePatientCommand parse(String args) throws ParseException {
         boolean isForceDelete = false;
         String indexToParse = args;
 
-        try {
-            if (forceDeleteMatcher.matches()) {
-                final String forceDelete = forceDeleteMatcher.group("forceDelete").trim();
-                final String patientIndex = forceDeleteMatcher.group("patientIndex");
-
-                if (forceDelete.equals("--force")) {
-                    isForceDelete = true;
-                    indexToParse = patientIndex;
-                }
+        if (forceDeleteMatcher.matches()) {
+            final String forceDelete = forceDeleteMatcher.group("forceDelete").trim();
+            final String patientIndex = forceDeleteMatcher.group("patientIndex");
+
+            if (forceDelete.equals("--force")) {
+                isForceDelete = true;
+                indexToParse = patientIndex;
             }
+        }
```

Comment:
actually, i feel like we should throw an exception if forceDelete not equals `"--force"` as well? feels like we shouldn't just let a user pass with an input such as `delete-patient aisjdhabakahbakshdbankdjashb 1`

if we do this we can then make the inner `if` conditional not nested, and more flat

same for `DeleteDoctorCommandParser`

Reply:
I don't think an exception is needed as `indexToParse = args` by default, so `aisjdhabakahbakshdbankdjashb 1` will be captured as a ParseException when `ParserUtil` calls `parseIndex` on it.

Reply:
ah i missed that out, yeah then it works perfectly